zephyr/src/supp_mgmt.c

```diff
@@ -0,0 +1,380 @@
+	int argc = len;
+	int ret = 0;
+
+	ret = cli_main (argc, argv);
+	
+	return ret;
+
+}
+
+static int wifi_supp_status(uint32_t mgmt_request,
+			     struct net_if *iface,
+			     void *data,
+			     size_t len)
+{
+	char ** argv = (char **)data;
+	int argc = len;
+	int ret = 0;
+	
+	ret = cli_main (argc, argv);
+	
+	return ret;
+
+}
+
+static int wifi_supp_bssid(uint32_t mgmt_request,
+			     struct net_if *iface,
+			     void *data,
+			     size_t len)
+{
+	char ** argv = (char **)data;
+	int argc = len;
+	int ret = 0;
+	
+	ret = cli_main (argc, argv);
+	
+	return ret;
+
+}
+
+static int wifi_supp_sta_autoconnect(uint32_t mgmt_request,
+			     struct net_if *iface,
+			     void *data,
+			     size_t len)
+{
+	char ** argv = (char **)data;
+	int argc = len;
+	int ret = 0;
+	
+	ret = cli_main (argc, argv);
+	
+	return ret;
+
+}
+
+static int wifi_supp_signal_poll(uint32_t mgmt_request,
+			     struct net_if *iface,
+			     void *data,
+			     size_t len)
+{
+	char ** argv = (char **)data;
+	int argc = len;
+	int ret = 0;
+	
+	ret = cli_main (argc, argv);
+	
+	return ret;
+
+}
+
+static int wifi_supp_connect(uint32_t mgmt_request,
+			     struct net_if *iface,
+			     void *data,
+			     size_t len)
+{
+	struct wifi_connect_req_params *params =
+		(struct wifi_connect_req_params *)data;
+#ifdef notyet
+	const struct device *dev = net_if_get_device(iface);
+	const struct img_rpu_dev_ops_zep *nvlsi_dev_ops = dev->api;
+#endif /* notyet */
+	struct wpa_ssid *ssid = NULL;
+	bool pmf = true;
+
+	if (!wpa_s_0) {
+		wpa_printf(MSG_ERROR, "%s: wpa_supplicant is not initialized, dropping connect\n",
+			   __func__);
+		return -1;
+	}
+	ssid = wpa_supplicant_add_network(wpa_s_0);
+	ssid->ssid = os_zalloc(sizeof(u8) * MAX_SSID_LEN);
+
+	memcpy(ssid->ssid, params->ssid, params->ssid_length);
+	ssid->ssid_len = params->ssid_length;
+	ssid->disabled = 1;
+	ssid->key_mgmt = WPA_KEY_MGMT_NONE;
+
+	wpa_s_0->conf->filter_ssids = 1;
+	wpa_s_0->conf->ap_scan= 1;
+
+	if (params->psk) {
+		// TODO: Extend enum wifi_security_type
+		if (params->security == 3) {
+			ssid->key_mgmt = WPA_KEY_MGMT_SAE;
+			str_clear_free(ssid->sae_password);
+			ssid->sae_password = dup_binstr(params->psk, params->psk_length);
+			if (ssid->sae_password == NULL) {
+				printk("%s:Failed to copy sae_password\n", __func__);
+				return -1;
+			}
+		} else {
+			if (params->security == 2)
+				ssid->key_mgmt = WPA_KEY_MGMT_PSK_SHA256;
+			else
+				ssid->key_mgmt = WPA_KEY_MGMT_PSK;
+			str_clear_free(ssid->passphrase);
+			ssid->passphrase = dup_binstr(params->psk, params->psk_length);
+			if (ssid->passphrase == NULL) {
+				printk("%s:Failed to copy passphrase\n", __func__);
+				return -1;
+			}
+		}
+
+		wpa_config_update_psk(ssid);
+
+		if (pmf)
+			ssid->ieee80211w = 1;
+
+	}
+
+	wpa_supplicant_enable_network(wpa_s_0, ssid);
+	wpa_supplicant_select_network(wpa_s_0, ssid);
+
+	return 0;
+}
+
+NET_MGMT_REGISTER_REQUEST_HANDLER(NET_REQUEST_SUPP_SCAN, wifi_supp_scan);
```

Comment:
Weren't we supposed to drop the net mgmt use?

Reply:
dropped